src/server/bitops_family.cc

```diff
@@ -57,14 +65,88 @@ constexpr int32_t GetByteIndex(uint32_t offset) noexcept {
   return offset / OFFSET_FACTOR;
 }
 
-uint8_t GetByteValue(const std::string& str, uint32_t offset) {
+uint8_t GetByteValue(std::string_view str, uint32_t offset) {
   return static_cast<uint8_t>(str[GetByteIndex(offset)]);
 }
 
 constexpr bool CheckBitStatus(uint8_t byte, uint32_t offset) {
   return byte & (0x1 << offset);
 }
 
+constexpr std::uint8_t CountBitsRange(std::uint8_t byte, std::uint8_t from, uint8_t to) {
+  int count = 0;
+  for (int i = from; i < to; i++) {
+    count += CheckBitStatus(byte, GetNormalizedBitIndex(i));
+  }
+  return count;
+}
+
+constexpr std::uint8_t BitOnCount(std::uint8_t byte) {
+  return NIBBLE_LOOKUP[byte & 0x0F] + NIBBLE_LOOKUP[byte >> 4];
+}
+
+std::uint32_t CountSetAllBytes(std::string_view at, int32_t start, int32_t end) {
+  if (start >= end || start < 0 || end < 0) {
+    return 0;
+  }
+  end = std::min(end, static_cast<int32_t>(at.size()));  // don't overflow
```

Comment:
it's more logical to cast `end` to size_t (because you checked it's not negative a line before) instead of casting size_t to `int32_t` and risk the overflow bug.

Reply:
or even better, avoid using signed ints if it's not applicable

Reply:
Changed to size_t for all internal functions